src/core/Akka/Util/Internal/Collections/ListSlice.cs

```diff
@@ -0,0 +1,182 @@
+            }
+
+            public T Current
+            {
+                get
+                {
+                    if (_current < _start) throw new InvalidOperationException("Enumeration not started.");
+                    if (_current >= _end) throw new InvalidOperationException("Enumeration ended.");
+                    return _array[_current];
+                }
+            }
+
+            object IEnumerator.Current => Current;
+
+            public void Dispose()
+            {
+                
+            }
+        }
+        
+        private readonly IReadOnlyList<T> _array;
+
+        public ListSlice(IReadOnlyList<T> array)
+        {
+           
+            if (array == null)
+                throw new ArgumentNullException(nameof(array));
+
+            _array = array;
+            Offset = 0;
+            Count = array.Count;
+        }
+        
+        public ListSlice(IReadOnlyList<T> array, int offset, int count)
+        {
+            if (array == null)
+                throw new ArgumentNullException(nameof(array));
+            if (offset < 0)
+                throw new ArgumentOutOfRangeException(nameof(offset), "Cannot be below zero.");
+            if (count < 0)
+                throw new ArgumentOutOfRangeException(nameof(count), "Cannot be below zero.");
+            
+            _array = array;
+            Offset = offset;
+            Count = count;
+        }
+
+        public int Offset { get; }
+
+        public IReadOnlyList<T> Array => _array;
+
+        public IEnumerator<T> GetEnumerator()
+        {
+            return new SliceEnumerator(this);
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        public void Add(T item)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public void Clear()
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public bool Contains(T item)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public void CopyTo(T[] array, int arrayIndex)
+        {
+            var n = 0;
+            foreach (var i in _array.Skip(Offset).Take(Count))
+            {
+                array[arrayIndex + n++] = i;
+            }
+        }
+
+        public bool Remove(T item)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public int Count { get; }
+
+        public bool IsReadOnly => true;
+        public int IndexOf(T item)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public void Insert(int index, T item)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public void RemoveAt(int index)
+        {
+            throw new System.NotImplementedException();
+        }
+
+        public T this[int index]
+        {
+            get => _array.ElementAt(Offset + index);
```

Comment:
Ah yes, that's a bit of leftovers from when I was using `IReadOnlyCollection` instead of `IReadOnlyList` - the former doesn't have an indexer. I'll update that.